Lib/unittest/case.py

```diff
@@ -353,15 +353,36 @@ def __exit__(self, exc_type, exc_value, tb):
         self.logger.handlers = self.old_handlers
         self.logger.propagate = self.old_propagate
         self.logger.setLevel(self.old_level)
+
         if exc_type is not None:
             # let unexpected exceptions pass through
             return False
+
+        self.check_records()
+
+
+class _AssertLogsContext(_AssertLogsBaseContext):
+    """A context manager used to implement TestCase.assertLogs()."""
+
+    def check_records(self):
         if len(self.watcher.records) == 0:
             self._raiseFailure(
                 "no logs of level {} or higher triggered on {}"
                 .format(logging.getLevelName(self.level), self.logger.name))
 
 
+class _AssertNoLogsContext(_AssertLogsBaseContext):
+    """A context manager used to implement TestCase.assertNoLogs()."""
+
+    def check_records(self):
+        if len(self.watcher.records) > 0:
+            self._raiseFailure(
+                "Logs unexpected found: {!r}".format(
```

Comment:
Thanks a lot for adding this! It's a good improvement. I think this failure message would be improved if it was `Unexpected logs found: ...`.

Reply:
Oops, yes you are right. Thank you for spotting this.